config/src/main/java/org/springframework/security/config/annotation/web/configurers/PermitAllSupport.java

```diff
@@ -48,11 +48,23 @@ static void permitAll(HttpSecurityBuilder<? extends HttpSecurityBuilder<?>> http
 			RequestMatcher... requestMatchers) {
 		ExpressionUrlAuthorizationConfigurer<?> configurer = http
 				.getConfigurer(ExpressionUrlAuthorizationConfigurer.class);
-		Assert.state(configurer != null, "permitAll only works with HttpSecurity.authorizeRequests()");
+		AuthorizeHttpRequestsConfigurer<?> httpConfigurer = http.getConfigurer(AuthorizeHttpRequestsConfigurer.class);
+
+		boolean oneConfigurerPresent = (configurer != null && httpConfigurer == null)
```

Comment:
Is this a good use case for `^`?

```suggestion
		boolean oneConfigurerPresent = configurer == null ^ httpConfigurer == null;
```